src/network/server.rs

```diff
@@ -302,98 +307,109 @@ impl MyHubService {
                     ));
                 }
 
-                let stores = self
-                    .get_stores_for(fid)
-                    .map_err(|_| HubError::internal_db_error("stores not found for fid"))?;
-
-                let id_register = stores
-                    .onchain_event_store
-                    .get_id_register_event_by_fid(fid)
-                    .map_err(|_| HubError::internal_db_error("Could not fetch id registration"))?;
-
-                match id_register {
-                    None => return Err(HubError::validation_failure("missing fid registration")),
-                    Some(id_register) => {
-                        match id_register.body {
-                            Some(Body::IdRegisterEventBody(id_register)) => {
-                                // Check verified addresses if the resolved address doesn't match the custody address
-                                if id_register.to != resolved_ens_address {
-                                    let verification = VerificationStore::get_verification_add(
-                                        &stores.verification_store,
-                                        fid,
-                                        &resolved_ens_address,
-                                    )?;
-
-                                    match verification {
-                                    None => Err(HubError::validation_failure("invalid ens proof, no matching custody address or verified addresses")),
-                                    Some(_) => Ok(()),
-                                }
-                                } else {
-                                    Ok(())
-                                }
-                            }
-                            _ => {
-                                return Err(HubError::validation_failure(
-                                    "missing fid registration",
-                                ))
-                            }
-                        }
-                    }
-                }
+                self.chain_clients.for_chain(Chain::BaseMainnet)?
             }
-        }
+            _ => {
+                return Err(HubError::validation_failure(
+                    format!(
+                        "unsupported username type: {} for name: {}",
+                        proof.r#type, name,
+                    )
+                    .as_str(),
+                ))
+            }
+        };
+
+        let resolved_ens_address = chain_api
+            .resolve_ens_name(name.to_string())
+            .await
+            .map_err(|err| {
+                HubError::validation_failure(
+                    format!("ENS resolution error: {}", err.to_string()).as_str(),
+                )
+            })?
+            .to_vec();
+
+        Ok(resolved_ens_address)
     }
 
     async fn validate_ens_username(&self, fid: u64, name: String) -> Result<(), HubError> {
```

Comment:
This function only queries for username proofs with `UsernameTypeEnsL1`. Not sure how the tests are passing but I think we also need to incorporate the basename type.